Fix part #17,#382: Hi fi topic audio bar

nikitamarysolomanpvt opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
Hi fi topic audio bar 

## High-fi Checklist
- [x] Seekbar and its animation
<img width="184" alt="Screenshot 2019-11-08 at 3 24 11 PM" src="https://user-images.githubusercontent.com/9396084/68467424-25947100-023c-11ea-89d2-74a745caaa35.png">

- [x] AudioBar along with Volume icon in toolbar
<img width="327" alt="Screenshot 2019-11-08 at 3 24 37 PM" src="https://user-images.githubusercontent.com/9396084/68467428-275e3480-023c-11ea-8915-0e8d074073ee.png">

## Issue Link
https://github.com/oppia/oppia-android/issues/17

## Mock Link
[https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/9ae3d04d-9845-4b75-abf8-8f69e8823fb1/PM-IN-AU-Playing-2-/](https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/9ae3d04d-9845-4b75-abf8-8f69e8823fb1/PM-IN-AU-Playing-2-/)

## GIF 

![new](https://user-images.githubusercontent.com/54615666/69333864-8df14280-0c7f-11ea-8bee-5e18f615cca0.gif)


## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

nikitamarysolomanpvt committed:
thumb seekbar

nikitamarysolomanpvt committed:
shadow to audio bar

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

veena14cs commented:
In Pixel 3a Xl device gap is too much between play button and seekbar and also between language. Please check in the screenshot.

![Screenshot_20191121-175159](https://user-images.githubusercontent.com/16301028/69337885-f5ab8b80-0c87-11ea-86f5-6e3abdc0e36e.png)

nikitamarysolomanpvt committed:
audio bar width as per screen width

veena14cs commented:
@mschanteltc can we have lighter background for audio bar or seekbar and thumb color lighter then the background. Because seekbar is not clearly visible on audiobar.

nikitamarysolomanpvt commented:
> In Pixel 3a Xl device gap is too much between play button and seekbar and also between language. Please check in the screenshot.
> 
> ![Screenshot_20191121-175159](https://user-images.githubusercontent.com/16301028/69337885-f5ab8b80-0c87-11ea-86f5-6e3abdc0e36e.png)

Done..

rt4914 reviewed (approved):
This looks good and discussed with @nikitamarysolomanpvt earlier, the audiobar does have issues, which will be picked up once we get clarifications from Chantel.

mschanteltc commented:
I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3. 
![Audio Scrub](https://user-images.githubusercontent.com/22177674/69371122-0e01b300-0c54-11ea-95bd-3adf3c65d169.png)

nikitamarysolomanpvt commented:
> I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.

Please see the drop-shadow 
opacity:54%
<img width="347" alt="small" src="https://user-images.githubusercontent.com/54615666/69404142-8171f600-0d22-11ea-9cbf-998c97889b0b.png">
drop-shadow opacity:20%
<img width="297" alt="small-2" src="https://user-images.githubusercontent.com/54615666/69403378-71f1ad80-0d20-11ea-9582-1cf1c30eef9b.png">
Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...

nikitamarysolomanpvt committed:
audio bar width as per screen width

mschanteltc commented:
> > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> 
> Please see the drop-shadow
> opacity:54%
> <img alt="small" width="347" src="https://user-images.githubusercontent.com/54615666/69404142-8171f600-0d22-11ea-9cbf-998c97889b0b.png">
> drop-shadow opacity:20%
> <img alt="small-2" width="297" src="https://user-images.githubusercontent.com/54615666/69403378-71f1ad80-0d20-11ea-9582-1cf1c30eef9b.png">
> Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...

I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.

nikitamarysolomanpvt commented:
> > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > 
> > 
> > Please see the drop-shadow
> > opacity:54%
> > <img alt="small" width="347" src="https://user-images.githubusercontent.com/54615666/69404142-8171f600-0d22-11ea-9cbf-998c97889b0b.png">
> > drop-shadow opacity:20%
> > <img alt="small-2" width="297" src="https://user-images.githubusercontent.com/54615666/69403378-71f1ad80-0d20-11ea-9582-1cf1c30eef9b.png">
> > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> 
> I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.

Ok then iam not using the svg image that i requested you.

mschanteltc commented:
> > > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > > 
> > > 
> > > Please see the drop-shadow
> > > opacity:54%
> > > <img alt="small" width="347" src="https://user-images.githubusercontent.com/54615666/69404142-8171f600-0d22-11ea-9cbf-998c97889b0b.png">
> > > drop-shadow opacity:20%
> > > <img alt="small-2" width="297" src="https://user-images.githubusercontent.com/54615666/69403378-71f1ad80-0d20-11ea-9582-1cf1c30eef9b.png">
> > > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> > 
> > 
> > I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.
> 
> Ok then iam not using the svg image that i requested you.

LGTM

veena14cs reviewed (approved):
LGTM Please check if it passes accessibility test

BenHenning commented on app/src/main/res/drawable/audio_seekbar_background.xml:
```diff
@@ -1,8 +1,20 @@
+      <shape>
+        <solid
+          android:color="@color/colorAccent"/>
+        <corners android:radius="7dp" />
```
Shouldn’t all dimensions be in multiples of 4dp per material guidelines?

nikitamarysolomanpvt replied:
Done

BenHenning reviewed (approved):
Thanks!

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt commented:
> > > > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > > > 
> > > > 
> > > > Please see the drop-shadow
> > > > opacity:54%
> > > > <img alt="small" width="347" src="https://user-images.githubusercontent.com/54615666/69404142-8171f600-0d22-11ea-9cbf-998c97889b0b.png">
> > > > drop-shadow opacity:20%
> > > > <img alt="small-2" width="297" src="https://user-images.githubusercontent.com/54615666/69403378-71f1ad80-0d20-11ea-9582-1cf1c30eef9b.png">
> > > > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> > > 
> > > 
> > > I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.
> > 
> > 
> > Ok then iam not using the svg image that i requested you.
> 
> LGTM

## Accessibility Scanner Results: Oppia - 2019-11-25-11:25:39

![screenshot_Oppia_2019-11-25-11_25_39](https://user-images.githubusercontent.com/54615666/69516616-2cc7b880-0f78-11ea-8072-232f81d14097.png)Item label
org.oppia.app:id/sbAudioProgress
This item may not have a label readable by screen readers.

Touch target
org.oppia.app:id/tvAudioLanguage

@mschanteltc Could you please suggest ...
This item's width is 46dp. Consider making the width of this touch target 48dp or larger.

nikitamarysolomanpvt commented:
> LGTM Please check if it passes accessibility test

Done

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt commented:
@rt4914 & @veena14cs i fixed #382 so PTAL

rt4914 commented:
@nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.

nikitamarysolomanpvt committed:
play and pause icons size

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

nikitamarysolomanpvt committed:
nit

veena14cs commented on app/src/main/res/drawable/audio_seekbar_background.xml:
```diff
@@ -1,8 +1,20 @@
+    <shape>
+      <solid
+        android:color="@color/white"/>
+
```
Remove this extra space.

veena14cs reviewed (approved):
LGTM one nit change.

rt4914 reviewed (approved):
LGTM

nikitamarysolomanpvt commented:
> @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.

@rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?

nikitamarysolomanpvt committed:
nit

rt4914 commented:
> > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> 
> @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?

I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.

nikitamarysolomanpvt commented:
> > > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> > 
> > 
> > @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?
> 
> I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.

@rt4914 I checked and it's working perfectly on both devices and emulator... PTAL

rt4914 commented:
> > > > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> > > 
> > > 
> > > @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?
> > 
> > 
> > I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.
> 
> @rt4914 I checked and it's working perfectly on both devices and emulator... PTAL

On further investigation it was found that it is failing on Android API 19.

rt4914 commented:
> > > > > @nikitamarysolomanpvt I tried to play the audio, the audio-bar is visible but it is not playable anymore. It maybe low-fi issue as well, but still needs some investigation.
> > > > 
> > > > 
> > > > @rt4914 Which topic you tried. Please be specific when you mention such issues... is this issue there for all topics or in specific topic?
> > > 
> > > 
> > > I checked on all topics and it was not working. But on giving a second look actually it was working on Real device and it was failing on Emulator. That's why I approved it in second pass.
> > 
> > 
> > @rt4914 I checked and it's working perfectly on both devices and emulator... PTAL
> 
> On further investigation it was found that it is failing on Android API 19.

I have filed this as an issue #475

mschanteltc commented:
> > > > > > I updated the colors of the seekbar so that the played time range is white and remaining time range is #2D4A9D with a "[multiply](https://en.wikipedia.org/wiki/Blend_modes#Multiply)" blend mode, resulting in a #081661 color. The audio thumb is also changed to #FFFFFF with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3, blur: 3.
> > > > > 
> > > > > 
> > > > > Please see the drop-shadow
> > > > > opacity:54%
> > > > > <img alt="small" width="347" src="https://user-images.githubusercontent.com/54615666/69404142-8171f600-0d22-11ea-9cbf-998c97889b0b.png">
> > > > > drop-shadow opacity:20%
> > > > > <img alt="small-2" width="297" src="https://user-images.githubusercontent.com/54615666/69403378-71f1ad80-0d20-11ea-9582-1cf1c30eef9b.png">
> > > > > Please let me know which one you prefer ... If you want me to use the image that you have provided .. i may use that also if this is not looking good...
> > > > 
> > > > 
> > > > I think the audio thumb with the 54% opacity is looking great! It's more evident that it appears above the scrub and has a "tappable" appearance.
> > > 
> > > 
> > > Ok then iam not using the svg image that i requested you.
> > 
> > 
> > LGTM
> 
> ## Accessibility Scanner Results: Oppia - 2019-11-25-11:25:39
> ![screenshot_Oppia_2019-11-25-11_25_39](https://user-images.githubusercontent.com/54615666/69516616-2cc7b880-0f78-11ea-8072-232f81d14097.png)Item label
> org.oppia.app:id/sbAudioProgress
> This item may not have a label readable by screen readers.
> 
> Touch target
> org.oppia.app:id/tvAudioLanguage
> 
> @mschanteltc Could you please suggest ...
> This item's width is 46dp. Consider making the width of this touch target 48dp or larger.

Hi Nikita,

I fixed the language selector button to be 48 px in width. LMK what you think.

nikitamarysolomanpvt commented:
Thats already done. I set it as 48dp

On Wed, 27 Nov 2019 at 12:46 AM, Chantel <notifications@github.com> wrote:

> I updated the colors of the seekbar so that the played time range is white
> and remaining time range is #2D4A9D with a "multiply
> <https://en.wikipedia.org/wiki/Blend_modes#Multiply>" blend mode,
> resulting in a #081661 color. The audio thumb is also changed to #FFFFFF
> with a drop shadow (#000000, opacity: 20%) with x-offset: 0, y-offset: 3,
> blur: 3.
>
> Please see the drop-shadow
> opacity:54%
> [image: small]
> <https://user-images.githubusercontent.com/54615666/69404142-8171f600-0d22-11ea-9cbf-998c97889b0b.png>
> drop-shadow opacity:20%
> [image: small-2]
> <https://user-images.githubusercontent.com/54615666/69403378-71f1ad80-0d20-11ea-9582-1cf1c30eef9b.png>
> Please let me know which one you prefer ... If you want me to use the
> image that you have provided .. i may use that also if this is not looking
> good...
>
> I think the audio thumb with the 54% opacity is looking great! It's more
> evident that it appears above the scrub and has a "tappable" appearance.
>
> Ok then iam not using the svg image that i requested you.
>
> LGTM
>
> Accessibility Scanner Results: Oppia - 2019-11-25-11:25:39
>
> [image: screenshot_Oppia_2019-11-25-11_25_39]
> <https://user-images.githubusercontent.com/54615666/69516616-2cc7b880-0f78-11ea-8072-232f81d14097.png>Item
> label
> org.oppia.app:id/sbAudioProgress
> This item may not have a label readable by screen readers.
>
> Touch target
> org.oppia.app:id/tvAudioLanguage
>
> @mschanteltc <https://github.com/mschanteltc> Could you please suggest ...
> This item's width is 46dp. Consider making the width of this touch target
> 48dp or larger.
>
> Hi Nikita,
>
> I fixed the language selector button to be 48 px in width. LMK what you
> think.
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia-android/pull/467?email_source=notifications&email_token=ANAV44XUPN6ZF4W5NTKB2UDQVVYZTA5CNFSM4JQA3XIKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFHEOFY#issuecomment-558778135>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANAV44RTJUMZ3ZXMVI43E3LQVVYZTANCNFSM4JQA3XIA>
> .
>
